samples/java-customer-registry/src/main/java/customer/CustomerValueEntityFactory.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package customer;
+
+import com.akkaserverless.javasdk.valueentity.ValueEntityContext;
+import com.akkaserverless.javasdk.valueentity.ValueEntityFactory;
+import com.akkaserverless.javasdk.valueentity.ValueEntityOptions;
+import com.google.protobuf.Descriptors;
+import customer.api.CustomerApi;
+
+public class CustomerValueEntityFactory implements ValueEntityFactory<CustomerValueEntity> {
+
+  @Override
+  public CustomerValueEntity newInstance(ValueEntityContext context) {
+    return new CustomerValueEntity(context);
+  }
+
+  @Override
+  public ValueEntityOptions options() {
+    return ValueEntityOptions.defaults();
+  }
+
+  // depends on proto file definition therefore final
+  @Override
+  public final Descriptors.ServiceDescriptor serviceDescriptor() {
+    return CustomerApi.getDescriptor().findServiceByName("CustomerService");
+  }
+
+  // depends on proto file definition therefore final
+  @Override
+  public final String entityTypeHint() {
+    return "customers";
+  }
+
+  // depends on proto file definition therefore final
+  @Override
+  public final CustomerValueEntityHandler newHandler(ValueEntityContext context) {
+    return new CustomerValueEntityHandler(newInstance(context));
+  }
+
+  // depends on proto file definition therefore final
+  @Override
+  public final Descriptors.FileDescriptor[] additionalDescriptors() {
+    return new Descriptors.FileDescriptor[0];
+  }
```

Comment:
This is marked final just as an indication that users should preferably not modify these method. However, since it will live in `src/main/java`, they will be able to modify them.

Or maybe we should have another intermediate class that lives under the `generated-sources` dir and defines these method. WDYT?